api/src/main/java/io/minio/MinioClient.java

```diff
@@ -686,11 +695,54 @@ private void checkBucketName(String name) throws InvalidBucketNameException {
    * @param readTimeout       HTTP read timeout in milliseconds.
    */
   public void setTimeout(long connectTimeout, long writeTimeout, long readTimeout) {
-    httpClient.setConnectTimeout(connectTimeout, TimeUnit.MILLISECONDS);
-    httpClient.setWriteTimeout(writeTimeout, TimeUnit.MILLISECONDS);
-    httpClient.setReadTimeout(readTimeout, TimeUnit.MILLISECONDS);
+    this.httpClient = this.httpClient.newBuilder()
+      .connectTimeout(connectTimeout, TimeUnit.MILLISECONDS)
+      .writeTimeout(writeTimeout, TimeUnit.MILLISECONDS)
+      .readTimeout(readTimeout, TimeUnit.MILLISECONDS)
+      .build();
   }
 
+
+  /**
+   * Ignores check on server certificate for HTTPS connection.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code minioClient.ignoreCertCheck(); }</pre>
+   *
+   */
+  public void ignoreCertCheck() throws NoSuchAlgorithmException, KeyManagementException {
+    final TrustManager[] trustAllCerts = new TrustManager[] { new X509TrustManager() {
+        @Override
+        public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        }
+
+        @Override
+        public X509Certificate[] getAcceptedIssuers() {
+          return new X509Certificate[]{};
+        }
+      }
+    };
+                                                             
+    final SSLContext sslContext = SSLContext.getInstance("SSL");
+    sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+    final SSLSocketFactory sslSocketFactory = sslContext.getSocketFactory();
+
+    this.httpClient = this.httpClient.newBuilder()
+      .sslSocketFactory(sslSocketFactory, (X509TrustManager)trustAllCerts[0])
+      .hostnameVerifier(new HostnameVerifier() {
+          @Override
+          public boolean verify(String hostname, SSLSession session) {
+            return true;
+          }
+        })
+      .build();
+  }
+  
+
```

Comment:
Currently apis doing ReST calls are documented into README.md/API.md and others are covered in javadoc.